### DISCUSSION TITLE
The GenericModel generated json schema seems to have an issue

### DISCUSSION THREAD

**Discussion by zaw007** [fastapi/sqlmodel#1002] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Generic, TypeVar

from pydantic import BaseModel
from sqlmodel import SQLModel, Field

__all__ = ['A', 'B', 'C']

SQLModelT = TypeVar('SQLModelT', bound=SQLModel)

class A(SQLModel, Generic[SQLModelT]):
    code: int
    b: SQLModelT


class B(BaseModel, Generic[SQLModelT]):
    code: int
    b: SQLModelT


class C(SQLModel):
    id: int
```

### Description

A using a generic SQLModel subclass to create，B using a generic BaseModel subclass to create, C is simple SQLModel.

The json schema generated by A[C] cannot be displayed correctly in the FastAPI /docs, the schema for C is not displayed. 

The B[C] is normal.
![image](https://github.com/tiangolo/sqlmodel/assets/25732226/8447c017-8ad2-4af4-b274-8776944efdcf)

### Operating System

Windows

### SQLModel Version

0.0.19

### Python Version

3.10.10

### Additional Context

pydantic 2.7.4
FastAPI 0.111.0

**Comment by ryangalamb** [comment]
It's worse than just the json schema. The typevar isn't used for validation at all when `SQLModel` is the base class.

I've opened a PR to fix the issue: https://github.com/fastapi/sqlmodel/pull/1275

In the meantime, I've worked around this the same way you described: by using `pydantic.BaseModel` instead of `SQLModel` for **just** the generic classes.

**Comment by YuriiMotov** [comment]
Even simpler case doesn't work:
```py
from typing import Generic, TypeVar

from fastapi import FastAPI
from sqlmodel import SQLModel

T = TypeVar('T')

class A(SQLModel, Generic[T]):
    code: int
    b: T

app = FastAPI()

@app.get("/a")
def get_a() -> A[int]:
    pass
```

<img width="253" height="164" alt="image" src="https://github.com/user-attachments/assets/fcd1ca68-c55d-4a03-87c4-b498b3973907" />

---

```json
      "A_int_": {
        "properties": {
          "code": {
            "type": "integer",
            "title": "Code"
          },
          "b": {
            "title": "B"
          }
        },
        "type": "object",
        "required": [
          "code",
          "b"
        ],
        "title": "A[int]"
      }
```
